misc/test-client/src/main.rs

```diff
@@ -41,22 +46,67 @@ struct CliArgs {
 
   #[clap(short, long, default_value = "30")]
   task_count: usize,
+
+  #[clap(long, help = "Omit threshold header in requests")]
+  omit_threshold_header: bool,
+}
+
+#[derive(Serialize)]
+struct RandomnessRequest {
+  points: Vec<String>,
+  epoch: u8,
 }
 
-fn generate_messages(layers: &[Vec<u8>], cli_args: &CliArgs, count: usize) -> Vec<String> {
+#[derive(Deserialize)]
+struct RandomnessResponse {
+  points: Vec<String>,
+}
+
+async fn generate_messages(layers: &[Vec<u8>], cli_args: &CliArgs, count: usize) -> Vec<String> {
   let rnd_fetcher = LocalFetcher::new();
   let example_aux = vec![];
-  (0..count)
-    .map(|_| {
-      let rrs = client::prepare_measurement(layers, cli_args.epoch).unwrap();
-      let req = client::construct_randomness_request(&rrs);
-
+  let client = reqwest::Client::new();
+
+  let mut messages = Vec::with_capacity(count);
+
+  for _ in 0..count {
+    let rrs = client::prepare_measurement(layers, cli_args.epoch).unwrap();
+    let req = client::construct_randomness_request(&rrs);
+
+    let points = if let Some(randomness_url) = &cli_args.randomness_server_url {
```

Comment:
The remote-request branch of the conditional here is pretty big. Might be better as a helper function.

Reply:
agreed